ERCS/erc-7774.md

```diff
@@ -0,0 +1,178 @@
+
+#### Cache Key-Value Mapping
+
+The `web3://` client maintains a key-value mapping for caching, which MUST be cleared whenever it transitions from "Listening for Events" to "Not Listening for Events." The mapping is structured as follows:
+
+```
+mapping(
+  (<chain id>, <contract address>, <ERC-6860 pathQuery>) 
+  => 
+  (<last modified date>, <ETag>)
+)
+```
+
+Additional elements can be included in the mapping key when necessary. For example, [ERC-7618](./eip-7618.md) requires the inclusion of the `Accept-Encoding` request header in the mapping key.
+
+#### Handling Requests in "Listening for Events" State
+
+When a request is received in the "Listening for Events" state:
+
+1. **If no mapping entry exists**:
+   - The `web3://` client queries the smart contract.
+   - If the response includes the `evm-events` directive in the `Cache-Control` header and an `ETag`, a mapping entry is created using the `ETag`.
+   - If the response contains the `evm-events` directive and a `max-age=XX` directive in the `Cache-Control` header, the mapping entry is created with the `last modified date`, determined in the following order of priority:
+     - The `Last-Modified` header, if present.
+     - The `Date` header, if present.
+     - Otherwise, the block date when the smart contract was queried.
+   - If the response includes both an `ETag` and a `Cache-Control: evm-events max-age=XX` directive, a single mapping entry is created containing both the `ETag` and the `last modified date`.
+
+2. **If a mapping entry exists**:
+   - If the request contains a valid `If-None-Match` header:
+     - If the `ETag` in the mapping matches the `If-None-Match` value, the `web3://` client returns a `304 Not Modified` response immediately.
+     - If the `ETag` does not match, the client queries the smart contract, deletes the mapping entry, and processes the request as if no mapping entry existed.
+   
+   - If the request contains a valid `If-Modified-Since` header:
+     - If the `last modified date` in the mapping is earlier than the `If-Modified-Since` date, the client returns a `304 Not Modified` response immediately.
+     - Otherwise, the client queries the smart contract, deletes the mapping entry, and processes the request as if no mapping entry existed.
+   
+   - If the request contains neither `If-None-Match` nor `If-Modified-Since` headers (or they are invalid):
+     - The client queries the smart contract, deletes the mapping entry, and processes the request as if no mapping entry existed.
+
+
+
+#### Cache Invalidation via Blockchain Events
+
+In the "Listening for Events" state, the `web3://` client listens to the blockchain for the cache invalidation events defined in the previous section. For each path match, it deletes the corresponding mapping entry.
+
+
+## Rationale
+
+We add this feature to the [ERC-6944](./eip-6944.md) resolve mode because it can be added without changes the interface.
```

Comment:
This line is more motivation than rationale.

Reply:
Thanks! The line between motivation and rationale is always a bit blurry :)